core/src/main/java/uk/ac/bbsrc/tgac/miso/core/manager/MisoFilesManager.java

```diff
@@ -186,11 +86,7 @@ public File generateTemporaryFile(String prefix, String suffix) throws IOExcepti
     return Collections.emptyList();
   }
 
-  public Collection<String> getFileNames(Object type, String qualifier) throws IOException {
-    return getFileNames(type.getClass(), qualifier);
-  }
-
-  protected File getFile(Class type, String qualifier, String fileName, boolean createIfNotExist) throws IOException {
+  protected File getFile(Class<?> type, String qualifier, String fileName, boolean createIfNotExist) throws IOException {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)